cmd/krel/cmd/release_notes.go

```diff
@@ -787,11 +899,393 @@ func (o *releaseNotesOptions) Validate() error {
+	if err != nil {
+		return errors.Wrap(err, "asking to retrieve last session")
+	}
+
+	// Bring up the provider
+	provider, err := notes.NewProviderFromInitString(workdir)
+	if err != nil {
+		return errors.Wrap(err, "while getting map provider for current notes")
+	}
+
+	const (
+		spacer = "    │ "
+		nl     = "\n"
+	)
+
+	// Cycle all gathered release notes
+	for pr, note := range releaseNotes.ByPR() {
+		if _, ok := pullRequestChecklist[pr]; ok && continueFromLastSession {
+			logrus.Debugf("Pull Request %d already reviewed", pr)
+			continue
+		}
+		title := fmt.Sprintf("Release Note for PR %d:", pr)
+		fmt.Println(nl + title)
+		fmt.Println(strings.Repeat("=", len(title)))
+		fmt.Printf("Pull Request URL: %skubernetes/kubernetes/pull/%d%s", github.GitHubURL, pr, nl)
+		noteMaps, err := provider.GetMapsForPR(pr)
+		if err != nil {
+			return errors.Wrapf(err, "while getting map for PR #%d", pr)
+		}
+
+		// Capture the original note values to compare
+		originalNote := &notes.ReleaseNote{
+			Text:           note.Text,
+			Author:         note.Author,
+			Areas:          note.Areas,
+			Kinds:          note.Kinds,
+			SIGs:           note.SIGs,
+			Feature:        note.Feature,
+			ActionRequired: note.ActionRequired,
+			Documentation:  note.Documentation,
+			ReleaseVersion: note.ReleaseVersion,
+		}
+
+		if noteMaps != nil {
+			fmt.Println("✨ Note contents are modified with a map")
+			for _, noteMap := range noteMaps {
+				if err := note.ApplyMap(noteMap); err != nil {
+					return errors.Wrapf(err, "applying notemap for PR #%d", pr)
+				}
+			}
+		}
+
+		fmt.Println(pointIfChanged("Author", note.Author, originalNote.Author), "@"+note.Author)
+		fmt.Println(pointIfChanged("SIGs", note.SIGs, originalNote.SIGs), note.SIGs)
+		fmt.Println(pointIfChanged("Kinds", note.Kinds, originalNote.Kinds), note.Kinds)
+		fmt.Println(pointIfChanged("Areas", note.Areas, originalNote.Areas), note.Areas)
+		fmt.Println(pointIfChanged("Feature", note.Feature, originalNote.Feature), note.Feature)
+		fmt.Println(pointIfChanged("ActionRequired", note.ActionRequired, originalNote.ActionRequired), note.ActionRequired)
+		fmt.Println(pointIfChanged("ReleaseVersion", note.ReleaseVersion, originalNote.ReleaseVersion), note.ReleaseVersion)
+		// TODO: Implement note.Documentation
+
+		// Wrap the note for better readability on the terminal
+		fmt.Println(pointIfChanged("Text", note.Text, originalNote.Text))
+		text := util.WrapText(note.Text, 80)
+		fmt.Println(spacer + strings.ReplaceAll(text, nl, nl+spacer))
+
+		choice, err := util.AskYorNorDefault(fmt.Sprintf("\n- Fix note for PR #%d?", note.PrNumber), false)
+		if err != nil {
+			// If the user cancelled with ctr+c exit and continue the PR flow
+			if err.(util.UserInputError).IsCtrlC() {
+				logrus.Info("Input cancelled, exiting edit flow")
+				return nil
+			}
+			return errors.Wrap(err, "while asking to edit release note")
```

Comment:
This only exits cleanly if you `Ctrl+C` while it's waiting for input? What if you happened to hit `Ctrl+C` while it wasn't? Would it be more robust to [use a goroutine to capture the signal interrupt instead](https://golangcode.com/handle-ctrl-c-exit-in-terminal/)?

Reply:
Yes, it only reacts when asking the user for input and it is built by handling the interrupt. Currently its part of #1454 
I'll move the signal handling code into the `util.Ask() ` function and use that function instead.